### PR TITLE
Clean up CommitRef and optimize Azure operations

### LINKED ISSUE #69: Use CommitRef as a param on interfaces
Instead of separate Name, CommitId params

### LINKED ISSUE #70: Optimize Azure Blob operations to use If-None-Match etag
For **immutable** objects, use phantom writes (don't write if it already exists)

### PR DESCRIPTION
**Pull request by grant-d**
- Fixes #69 
- Fixes #70
Also add `DebuggerDisplay` to all suitable objects

### CODE REVIEW

File: src/SourceCode.Chasm.Repository.AzureBlob/AzureBlobChasmRepo.CommitRef.cs [hunk 1]
```diff
@@ -63,49 +63,50 @@ private async ValueTask<(CommitRef, AccessCondition)> ReadCommitRefImplAsync(Clo
...
-            if (existingCommitRef != CommitRef.Empty)
+            if (existingCommitId != CommitId.Empty)
             {
                 // Idempotent
-                if (existingCommitRef == commitRef)
+                if (existingCommitId == commitRef.CommitId)
                     return;
 
                 // Optimistic concurrency check
                 if (previousCommitId.HasValue
```
Comment (jcdickinson) [inline src/SourceCode.Chasm.Repository.AzureBlob/AzureBlobChasmRepo.CommitRef.cs]: Move above the `if` above this `if`. E.g. Someone might revert just after you checked in a change.

File: src/SourceCode.Chasm.Repository.AzureBlob/AzureBlobChasmRepo.CommitRef.cs [hunk 2]
```diff
@@ -63,49 +63,50 @@ private async ValueTask<(CommitRef, AccessCondition)> ReadCommitRefImplAsync(Clo
...
+            var blobName = DeriveCommitRefBlobName(branch, commitRef.Name);
             var blobRef = refsContainer.GetAppendBlobReference(blobName);
 
-            // Load existing commit ref in order to use its ETAG
-            var (existingCommitRef, accessCondition) = await ReadCommitRefImplAsync(blobRef, cancellationToken).ConfigureAwait(false);
+            // Load existing commit ref in order to use its etag
+            var (existingCommitId, ifMatchCondition) = await ReadCommitRefImplAsync(blobRef, cancellationToken).ConfigureAwait(false);
 
-            if (existingCommitRef != CommitRef.Empty)
+            if (existingCommitId != CommitId.Empty)
```
Comment (jcdickinson) [inline src/SourceCode.Chasm.Repository.AzureBlob/AzureBlobChasmRepo.CommitRef.cs]: Remove this condition. You must use an empty commit ID if the previous commit ID is empty.

File: src/SourceCode.Chasm.Repository.AzureTable/AzureTableChasmRepo.CommitRef.cs [hunk 3]
```diff
@@ -55,51 +59,53 @@ private static async ValueTask<(CommitRef, string)> ReadCommitRefImplAsync(Cloud
...
             var refsTable = _refsTable.Value;
-            var operation = DataEntity.BuildReadOperation(branch, name);
+            var operation = DataEntity.BuildReadOperation(branch, commitRef.Name);
 
-            // Load existing commit ref in order to use its ETAG
+            // Load existing commit ref in order to use its etag
             var (existingCommitId, etag) = await ReadCommitRefImplAsync(refsTable, operation, Serializer, cancellationToken).ConfigureAwait(false);
 
-            if (existingCommitId != CommitRef.Empty)
+            if (existingCommitId != CommitId.Empty)
```
Comment (jcdickinson) [inline src/SourceCode.Chasm.Repository.AzureTable/AzureTableChasmRepo.CommitRef.cs]: Same tweaks as above.

File: src/SourceCode.Chasm.Repository.AzureBlob/AzureBlobChasmRepo.CommitRef.cs (hunk 2 above)
Comment (grant-d) [reply src/SourceCode.Chasm.Repository.AzureBlob/AzureBlobChasmRepo.CommitRef.cs]: No that check tells us whether or not the CommitRef was found in storage at all. But I agree it's ambiguous, will change the pattern

File: src/SourceCode.Chasm.Repository.AzureBlob/AzureBlobChasmRepo.CommitRef.cs (hunk 1 above)
Comment (grant-d) [reply src/SourceCode.Chasm.Repository.AzureBlob/AzureBlobChasmRepo.CommitRef.cs]: Done

File: src/SourceCode.Chasm.Repository.AzureTable/AzureTableChasmRepo.CommitRef.cs (hunk 3 above)
Comment (grant-d) [reply src/SourceCode.Chasm.Repository.AzureTable/AzureTableChasmRepo.CommitRef.cs]: Done

**Pull request merged**